pennylane/interfaces/jax_jit_tuple.py

```diff
@@ -0,0 +1,297 @@
+import jax
+import jax.numpy as jnp
+
+import pennylane as qml
+from pennylane.interfaces import InterfaceUnsupportedError
+from pennylane.interfaces.jax import _compute_jvps
+from pennylane.interfaces.jax_jit import _validate_jax_version, _numeric_type_to_dtype
+
+dtype = jnp.float64
+
+
+def _tapes_shape_dtype_tuple(tapes, device):
+    """Auxiliary function for defining the jax.ShapeDtypeStruct objects given
+    the tapes and the device.
+
+    The jax.pure_callback function expects jax.ShapeDtypeStruct objects to
+    describe the output of the function call.
+    """
+    shape_dtypes = []
+
+    def process_single_shape(shape, tape_dtype):
+        return jax.ShapeDtypeStruct(tuple(shape), tape_dtype)
+
+    for t in tapes:
+        num_measurements = len(t.measurements)
+
+        shape = t.shape(device)
+        if num_measurements == 1:
+            tape_dtype = _numeric_type_to_dtype(t.numeric_type)
+            shape_and_dtype = process_single_shape(shape, tape_dtype)
+        else:
+            tape_dtype = tuple(_numeric_type_to_dtype(elem) for elem in t.numeric_type)
+            shape_and_dtype = tuple(process_single_shape(s, d) for s, d in zip(shape, tape_dtype))
+
+        shape_dtypes.append(shape_and_dtype)
+    return shape_dtypes
+
+
+def _jac_shape_dtype_tuple(tapes, device):
+    """Auxiliary function for defining the jax.ShapeDtypeStruct objects given
+    the tapes and the device.
+
+    The jax.pure_callback function expects jax.ShapeDtypeStruct objects to
+    describe the output of the function call.
+    """
+    shape_dtypes = []
+
+    for t in tapes:
+        num_measurements = len(t.measurements)
+
+        shape = t.shape(device)
+        if num_measurements == 1:
+            tape_dtype = _numeric_type_to_dtype(t.numeric_type)
+            shape_and_dtype = jax.ShapeDtypeStruct(tuple(shape), tape_dtype)
+        else:
+            tape_dtype = tuple(_numeric_type_to_dtype(elem) for elem in t.numeric_type)
+            shape_and_dtype = tuple(
+                jax.ShapeDtypeStruct(tuple(s), d) for s, d in zip(shape, tape_dtype)
+            )
+
+        if len(t.trainable_params) == 1:
+            shape_dtypes.append(shape_and_dtype)
+        else:
+            if num_measurements == 1:
+                s = [shape_and_dtype for _ in range(len(t.trainable_params))]
+                shape_dtypes.append(tuple(s))
+            else:
+                s = [tuple(_s for _ in range(len(t.trainable_params))) for _s in shape_and_dtype]
+                shape_dtypes.append(tuple(s))
+
+    if len(tapes) == 1:
+        return shape_dtypes[0]
+
+    return tuple(shape_dtypes)
+
+
+def _squeeze_nested(jvps, multi_measurements, multi_params):
+    """Post-process the JVP by squeezing arrays in the result with a nested structure."""
+    res_jvps = []
+    for m, p, j in zip(multi_measurements, multi_params, jvps):
+        if m and p:
+            res_jvps.append(tuple(jnp.squeeze(j_comp) for j_comp in j))
```

Comment:
Do we have some unecessary dimension created in the JVP like e.g. (3, 1) instead of (3)

Reply:
I could remove this squeezing post-processing, it was required with earlier solutions. Thanks for the question!